tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -96,6 +92,98 @@ internal readonly partial struct SecurityCoordinator
 
     internal static Span TryGetRoot(Span span) => span.Context.TraceContext?.RootSpan ?? span;
 
+    public IResult? RunWafForUser(string? userId = null, string? userLogin = null, string? userSessionId = null, bool fromSdk = false, Dictionary<string, string>? otherTags = null)
+    {
+        if (_httpTransport.IsBlocked)
+        {
+            return null;
+        }
+
+        IResult? result = null;
+        Dictionary<string, object>? addresses = null;
+        try
+        {
+            var additiveContext = GetOrCreateAdditiveContext();
```

Comment:
_Can_ this return `null`? Should it?

Reply:
Yes if we're not in a web context, it will return null.. as additive context is stored in `HttpContext`, so in case of a null result we bail out